activerecord/lib/active_record/insert_all.rb

```diff
@@ -198,7 +198,7 @@ def updatable_columns
         def touch_model_timestamps_unless(&block)
           model.timestamp_attributes_for_update_in_model.filter_map do |column_name|
             if touch_timestamp_attribute?(column_name)
-              "#{column_name}=(CASE WHEN (#{updatable_columns.map(&block).join(" AND ")}) THEN #{model.quoted_table_name}.#{column_name} ELSE CURRENT_TIMESTAMP END),"
+              "#{column_name}=(CASE WHEN (#{updatable_columns.map(&block).join(" AND ")}) THEN #{model.quoted_table_name}.#{column_name} ELSE #{quoted_current_timestamp_for(column_name)} END),"
```

Comment:
It occurs to me that every time we call this method, we would be using a slightly different `Time.now`. While unlikely, this could lead to something like

```sql
updated_at=(... ELSE 'fraction of a second before midnight' END),
updated_on=(... ELSE 'the next day' END),
```

I wonder if it makes sense to memoize the current time. The simplest thing would probably be to record the current time in `initialize`, since all usages of this class are of the form `InsertAll.new(...).execute`, so we don't have to worry about stale timestamps.

On a separate branch, I am looking into if it is possible to set timestamps automatically on create, so I'll likely need to memoize the time anyway (to ensure every inserted row has the same timestamps).